sympy/solvers/solvers.py

```diff
@@ -1908,15 +1919,16 @@ def key(sym):
                 rv = sorted(rv, key=key)
             return rv
 
-        solved_syms = set(solved_syms)  # set of symbols we have solved for
         legal = set(symbols)  # what we are interested in
         # sort so equation with the fewest potential symbols is first
         u = Dummy()  # used in solution checking
+        solved_syms = set()  # ignore for sake of ordering
```

Comment:
Why is this set before the loop if each iteration resets it anyway?

Reply:
Because it is needed to get the loop going with the call to `_ok_syms`. The appearance inside the loop near the top is (to me) preferred over hiding it at the bottom after all the if blocks are done.